File: cryptography-cipher/src/test/java/dev/retrotv/crypto/cipher/generator/KeyGeneratorTest.java
package dev.retrotv.crypto.cipher.generator;

import dev.retrotv.crypto.cipher.enums.ECipher;
import dev.retrotv.crypto.exception.GenerateException;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class KeyGeneratorTest {
    @Test
    @DisplayName("Exception 발생 테스트")
    void test_generateException() {
        assertThrows(IllegalArgumentException.class, () -> KeyGenerator.generateKey(4));
        assertThrows(GenerateException.class, () -> KeyGenerator.generateKey(ECipher.AES));
        assertThrows(GenerateException.class, () -> KeyGenerator.generateKey(ECipher.ARIA));
        assertThrows(GenerateException.class, () -> KeyGenerator.generateKey(ECipher.LEA));
    }

    @Test
    @DisplayName("key 생성 테스트")
    void test_generateKey() {
        byte[] key = KeyGenerator.generateKey(ECipher.AES, 16);
        assertEquals(16, key.length);

        key = KeyGenerator.generateKey(ECipher.DES);
        assertEquals(8, key.length);

        key = KeyGenerator.generateKey(ECipher.SEED);
        assertEquals(16, key.length);
    }
}

File: cryptography-chiper/src/test/kotlin/dev/retrotv/crypto/encryption/generator/KeyGeneratorTest.kt
package dev.retrotv.crypto.encryption.generator

import dev.retrotv.crypto.enums.ECipher.*
import dev.retrotv.crypto.exception.GenerateException
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Assertions.assertThrows
import kotlin.test.Test

class KeyGeneratorTest {

    @Test
    @DisplayName("Exception 발생 테스트")
    fun test_generateException() {
        assertThrows(IllegalArgumentException::class.java) {
            generateKey(4)
        }

        assertThrows(GenerateException::class.java) {
            generateKey(AES)
        }

        assertThrows(GenerateException::class.java) {
            generateKey(ARIA)
        }

        assertThrows(GenerateException::class.java) {
            generateKey(LEA)
        }
    }

    @Test
    @DisplayName("key 생성 테스트")
    fun test_generateKey() {
        var key = generateKey(AES, 16)
        assert(key.size == 16)

        key = generateKey(DES)
        assert(key.size == 8)

        key = generateKey(SEED)
        assert(key.size == 16)
    }
}